graphql.go

```diff
@@ -144,14 +144,67 @@ type Response struct {
 	Extensions map[string]interface{} `json:"extensions,omitempty"`
 }
 
+func (s *Schema) ModifyingFields(queryString string, operationName string, variables map[string]interface{}) ([]*errors.QueryError, []string) {
+	doc, qErr := query.Parse(queryString)
+	if qErr != nil {
+		return []*errors.QueryError{qErr}, nil
+	}
+
+	if err := validation.Validate(s.schema, doc, variables, s.maxDepth); err != nil {
+		return err, nil
+	}
+
+	op, err := getOperation(doc, operationName)
+	if err != nil {
+		return []*errors.QueryError{errors.Errorf("%s", err)}, nil
+	}
+
+	if op.Type == query.Mutation {
+		if _, ok := s.schema.EntryPoints["mutation"]; !ok {
+			return []*errors.QueryError{{Message: "no mutations are offered by the schema"}}, nil
+		}
+	}
+
+	fields := []string{}
+	mainField := op.Selections[0].(query.Field)
+	for _, arg := range mainField.Arguments {
+		fieldName := arg.Name.Name + "."
+		fields = append(fields, fieldsFromLiteral(fieldName, arg.Value)...)
+	}
+
+	return nil, nil
```

Comment:
You don't seem to be using the `fields` slice. Am I missing something? Why do you return `nil, nil`?